metaconfig-core/shared/src/main/scala/metaconfig/generic/Settings.scala

```diff
@@ -15,6 +16,32 @@ final class Settings[T](
     val settings: List[Setting],
     val annotations: List[StaticAnnotation]
 ) {
+  locally {
+    val map = new mutable.HashMap[String, Setting]
+    settings.foreach { x =>
+      val y = map.getOrElseUpdate(x.name, x)
+      require(y eq x, s"Multiple fields with name: '${x.name}'")
```

Comment:
Would it be possible to turn this into a ConfError instead of throwing an exception? I don’t enjoy using libraries that perform exception throwing validation inside constructors. It’s better to provide a validation method that users can optionally call instead.

Reply:
@olafurpg will do. however, optionally calling will lead to the same problem.

i wonder if there's a compile time check we can perform in the macros? those do throw if implicit is undefined or type is not case class.

Reply:
Can you elaborate on the problem these exceptions solve? 

It might be possible to extend the macro to report errors on duplicate names in some cases. Annotation parameters can be arbitrary expressions so their value may not be fully known until runtime.

Reply:
@olafurpg in the attached scalameta/scalafmt#2588, i was fixing a bug where i copy-pasted a deprecation annotation and forgot to fix the deprecated name.

before, i didn't really understand why the name was needed because the field name is also known (thought it was to display some sort of a message), so didn't pay too much attention.

turned out, it effectively disabled one of the parameters and pointed its actual name to the other one.

so, what's the best way to surface this, either at compile or at runtime?

(also, for ExtraName and DeprecatedName, their parameters are strings; are you saying they might not be evaluated until runtime?)

Reply:
moved from constructor into the companion apply.

Reply:
It’s fine to throw an exception. Don’t mind my original comment